packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -282,23 +282,32 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
 
     tsTokenCanFollowModifier() {
       return (
-        (this.match(tt.bracketL) ||
-          this.match(tt.braceL) ||
-          this.match(tt.star) ||
-          this.match(tt.ellipsis) ||
-          this.match(tt.privateName) ||
-          this.isLiteralPropertyName()) &&
-        !this.hasPrecedingLineBreak()
+        this.match(tt.bracketL) ||
+        this.match(tt.braceL) ||
+        this.match(tt.star) ||
+        this.match(tt.ellipsis) ||
+        this.match(tt.privateName) ||
+        this.isLiteralPropertyName()
       );
     }
 
+    tsNextTokenOnSameLineAndCanFollowModifier() {
+      this.next();
+      if (this.hasPrecedingLineBreak()) {
+        return false;
+      }
+      return this.tsTokenCanFollowModifier();
```

Comment:
This refactor can be done separately from this PR, but given the deleted comment, we could:
- use `hasFollowingLineBreak` instead of `hasPrecedingLineBreak`
- use `lookaheadCharCode` instead of `tsTokenCanFollowModifier`
- do not use `tsTryParse` to call `tsNextTokenCanFollowModifier`